core/frontend/src/test/tile/GltfReader.test.ts

```diff
@@ -977,96 +1249,148 @@ describe("GltfReader", () => {
-              propertyCount1++;
-            } else if(key === "UINT8_MAT2_VALUES") {
-              expect(compareArrays(array, expectedValuesVec4[instanceFeatures[entryCount]])).toBe(true);
-              propertyCount1++;
-            } else if(key === "UINT8_MAT3_VALUES") {
-              expect(compareArrays(array, expectedValuesMat3[instanceFeatures[entryCount]])).toBe(true);
-              propertyCount1++;
-            }else if(key === "UINT8_MAT4_VALUES") {
-              expect(compareArrays(array, expectedValuesMat4[instanceFeatures[entryCount]])).toBe(true);
-              propertyCount1++;
+          // Expect empty property set for null feature id = 3
+          if(instanceFeatures[entryCount] === 3){
+            expect(JSON.stringify(entry.properties.propertySet1) === JSON.stringify({})).toBe(true);
+          } else {
+            let propertyCount1 = 0;
+            for(const [key, value] of Object.entries(entry.properties.propertySet1)){
+              const array = value as any[];
+              if(key === "UINT8_VEC2_VALUES") {
+                expect(compareArrays(array, expectedValuesVec2[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              } else if(key === "UINT8_VEC3_VALUES") {
+                expect(compareArrays(array, expectedValuesVec3[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              } else if(key === "UINT8_VEC4_VALUES") {
+                expect(compareArrays(array, expectedValuesVec4[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              } else if(key === "UINT8_MAT2_VALUES") {
+                expect(compareArrays(array, expectedValuesVec4[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              } else if(key === "UINT8_MAT3_VALUES") {
+                expect(compareArrays(array, expectedValuesMat3[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              }else if(key === "UINT8_MAT4_VALUES") {
+                expect(compareArrays(array, expectedValuesMat4[instanceFeatures[entryCount]])).toBe(true);
+                propertyCount1++;
+              }
             }
+            expect(propertyCount1 === 6).toBe(true);
           }
-          expect(propertyCount1 === 6).toBe(true);
+
+          entryCount ++;
         }
+        expect(entryCount === 4).toBe(true);
+      });
+    });
 
-        entryCount ++;
-      }
-      expect(entryCount === 4).toBe(true);
+    describe("EXT_mesh_features", () => {
+      it("feature indices should map to correct vertices", async () => {
+        const elementIdToStructuralMetadataMap = new BatchedTileIdMap(iModel);
+
+        const reader = createReader(meshFeaturesExt, elementIdToStructuralMetadataMap)!;
+        expect(reader).toBeDefined();
+
+        const result = await reader.read();
+        expect(result).toBeDefined();
```

Comment:
A more robust test would verify that the features are actually present in `result`. Your test currently only verifies that the reader parsed them.

Reply:
Extended the tests case.